### ISSUE TITLE
Makefile fails

### ISSUE THREAD

**Issue by labrax** [datacarpentry/R-ecology-lesson#388]
Hi,

After cloning the project folder and doing a make I got the following error:

-----------------------
"(...)
label: unnamed-chunk-10 (with options) 
List of 1
 $ child: chr "_page_built_on.Rmd"

  |.................................................................| 100%
   inline R code fragments

/usr/bin/pandoc +RTS -K512m -RTS 00-before-we-start.utf8.md --to html --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash --output 00-before-we-start.html --smart --email-obfuscation none --standalone --section-divs --table-of-contents --toc-depth 3 --variable toc_float=1 --variable toc_selectors=h1,h2,h3 --variable toc_smooth_scroll=1 --variable toc_print=1 --template /home/vroth/R/x86_64-pc-linux-gnu-library/3.4/rmarkdown/rmd/h/default.html --highlight-style pygments --css style.css --include-before-body /tmp/RtmpgkNmas/rmarkdown-str17873ffec255.html --variable navbar=1 --variable body_padding=51 --variable header_padding=56 --variable 'theme:bootstrap' --include-in-header /tmp/RtmpgkNmas/rmarkdown-str17872720f2c4.html --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --include-in-header header.html --include-after-body footer.html 
Error in parse_block(g[-1], g[1], params.src) : duplicate label '_answer'
Calls: <Anonymous> ... process_file -> split_file -> lapply -> FUN -> parse_block
Execution halted
Makefile:7: recipe for target 'pages' failed
make: *** [pages] Error 1
-----------------------

My pandoc version is: pandoc 1.19.2.1 Compiled with pandoc-types 1.17.0.5, texmath 0.9.4.1, skylighting 0.3.3.1

Victor

**Comment by fmichonneau** [comment]
Hi Victor,

Is your repository up to date? Did you edit the content of the lesson? This shouldn't happen, and currently the lesson is building without any error on Travis.

**Comment by labrax** [comment]
Hi François,

I just cloned it today. I did not edit anything. Is there a check on
dependencies to install?

**Comment by fmichonneau** [comment]
You can try to run:

```r
remotes::install_deps()
```

from the root folder of the lesson in case it's an issue with dependencies. You'll need to install the `remotes` package first if you don't already have it (it's a `devtools` dependency).

The error message you get leaves me confused though. There is no indication of an `_answer` chunk or identifier in the 00-before-we-start.Rmd file, so I don't know why you'd get this error message.

**Comment by labrax** [comment]
The issue has been solved. It is necessary to install the dependencies. I did it with:
`devtools::install_deps()`
Unfortunately, I could not find this information in the readme. Is it available? Shall we add it with other information on devtools, pandoc dependencies?

**Comment by fmichonneau** [comment]
OK, thanks for letting us know. Do you which package was missing? (I'm curious how this error was generated).

I'll add it to the Makefile to ensure that all dependencies get pulled in during the building of the lessons.

**Comment by labrax** [comment]
i did not have devtools, neither lots of dependencies, as:

`dbplyr_1.2.1.tar.gz
assertthat_0.2.0.tar.gz
DBI_0.8.tar.gz
rlang_0.2.0.tar.gz
tibble_1.4.2.tar.gz
pillar_1.2.1.tar.gz
tidyselect_0.2.4.tar.gz
Rcpp_0.12.16.tar.gz
knitr_1.20.tar.gz
stringr_1.3.0.tar.gz
stringi_1.1.7.tar.gz
yaml_2.1.18.tar.gz
remotes_1.1.1.tar.gz
rmarkdown_1.9.tar.gz
tidyverse_1.2.1.tar.gz
forcats_0.3.0.tar.gz
haven_1.1.1.tar.gz
hms_0.4.2.tar.gz
lubridate_1.7.3.tar.gz
modelr_0.1.1.tar.gz
lazyeval_0.2.1.tar.gz
tidyr_0.8.0.tar.gz
readxl_1.0.0.tar.gz
cellranger_1.1.0.tar.gz
rematch_1.0.1.tar.gz
reprex_0.1.2.tar.gz
callr_2.0.2.tar.gz
rvest_0.3.2.tar.gz
selectr_0.3-2.tar.gz
xml2_1.2.0.tar.gz`

pandoc I installed using apt-get
`apt-get install pandoc`

**Comment by fmichonneau** [comment]
Thanks for your input @labrax. I added a step in the build process that ensures that all R packages needed get installed (see commit associated with #389). I'm going to close this issue, but let us know if you have any problem.
